### ISSUE TITLE
5.0.0 testing bugs

### ISSUE THREAD

**Issue by bullseye555** [rollerderby/scoreboard#505]
Found the following bugs within the 5.0.0 release after doing a mock-game (only about 15 min halves - I'd meant to get around to doing a full end-to-end test like this as part of the Beta, but unfortunately didn't get a chance)
I've left this game untouched on my system, and can PM anyone the external IP & Port if you want to jump in to the scoreboard [you can find me at https://www.facebook.com/nabro/ ]

Bug 1:
1. Create teams in the _Data Management_ page with Logo, League, Team Name, Colours and Skaters
![image](https://user-images.githubusercontent.com/28804763/164878358-7a7fdd4d-f920-4324-bb9b-1cfad3ab9223.png)
2. Start a New Game (then _Go To Current Game_)
![image](https://user-images.githubusercontent.com/28804763/164878378-7119d724-a472-418a-96cf-edc16cb3c168.png)
3. No Players or Team details load in
![image](https://user-images.githubusercontent.com/28804763/164878399-78f93742-ef02-4d30-97b2-607a945f1652.png)
![image](https://user-images.githubusercontent.com/28804763/164878403-24c037c9-bf2c-4276-b896-47b959a02365.png)
4. Clicking on _Sync Changes_ does nothing
5. Have to click in the Team drop-down, change the team to another one (which them imports _that_ team) and then revert the team back to the correct one
![image](https://user-images.githubusercontent.com/28804763/164878463-84a7c5be-3f9f-410f-9651-b24056bab3ac.png)
![image](https://user-images.githubusercontent.com/28804763/164878471-637376dc-30d7-42e6-9d55-73430f94326b.png)
Bug 1b:
After correcting the teams in Step 5 above, the _Uniform Colours_ changes from a drop-down to a free-text field only [see differences between the 2nd screenshot at Step 3 and the second at Step 5]. This doesn't change if the _Sync Changes_ is left unselected

Bug 2:
When importing saved teams from 4.1.4, the _Bench Alt _Captain_ and _Alt Captain_ aren't honoured
4.1.4 original: 
![image](https://user-images.githubusercontent.com/28804763/164884585-1db6faa9-3ee6-47f9-a844-17b58d035d21.png)
5.0.0 imported:
![image](https://user-images.githubusercontent.com/28804763/164884606-874c3445-af8b-4fda-b7c5-a738a0aecbf7.png)
JSON (as text)
[teams (4).json.txt](https://github.com/rollerderby/scoreboard/files/8546438/teams.4.json.txt)

Bug 3:
Non-skating players are shown as selectable
![image](https://user-images.githubusercontent.com/28804763/164878572-dae36fc7-db36-4df3-9f8d-6b8ba612b65f.png)
I've also confirmed these show in the _Penalty Tracking_, _Lineup Tracking_, and _Penalty + Lineup_ tracking fields

Bug 4a:
_Bench Alt Captain_ flag when adding a skater gets dropped
![image](https://user-images.githubusercontent.com/28804763/164878696-20f6c35d-d310-4775-9f9f-97a436727c19.png)
![image](https://user-images.githubusercontent.com/28804763/164878698-0323fda1-e824-45fc-94ed-e656742a2a92.png)
It can be re-added after the entry is made and doesn't get dropped
![image](https://user-images.githubusercontent.com/28804763/164878712-6a806332-08e2-4cbc-be17-40238b52bedf.png)
Bug 4b:
_Bench Staff_ only shows as a selectable Flag once the 'Skater' has been added
![image](https://user-images.githubusercontent.com/28804763/164878762-39abe4a8-66a0-440a-b2c8-b10786a675db.png)
![image](https://user-images.githubusercontent.com/28804763/164878774-3e37353f-37d1-4ec6-81c7-a89045802030.png)

Bug 5: 
Official Review TO not being highlighted on Operator Panel and the _dot_ isn't flashing
![image](https://user-images.githubusercontent.com/28804763/164882905-03d4c149-0f3f-4edc-9f6a-a13e8c6a30ad.png)
In this specific screenshot, it was a _Retained_ OR and the + isn't flashing, but in the original the Dot didn't flash either
I changed OR teams (simulating LEFT team finishing their OR and RIGHT team starting one immediately after) and grabbed a screen record to show the display
https://user-images.githubusercontent.com/28804763/164883089-17ee1d9e-fcc2-4aec-85b3-bc1e3d9fb2ab.mp4

Bug 6:
During _Line Up_ the same player can be selected for both Jammer and Pivot on the SBO Panel. Confirming in Lineup Panels (both LT & PLT) this is not possible and only reflects the double selection as being the Jammer
![image](https://user-images.githubusercontent.com/28804763/164883361-dc00c126-8b30-46c6-a0bc-38e6a3238bbb.png)
![image](https://user-images.githubusercontent.com/28804763/164883556-36b6965e-311b-481b-97dd-23d38a6f8a99.png)

Bug 7a:
Unstart Jam when AutoStart is set to _Jam_ results in a loop that just auto-restarts the jam, and eventually the _Unstart Jam_ button becomes unavailable
https://user-images.githubusercontent.com/28804763/164883452-ec0fad59-c0df-46ad-ba35-a16105d5f900.mp4
Recommend that when _Unstart Jam_ is used when _AutoStart = Jam_ then immediately enter an OTO

Bug 7b:
Unstart Jam has also twice resulted in a large amount of blank jams being added to the Scorekeeper Paperwork section - I wasn't able to replicate the exact process of how this happened. The first time (creating jams 9 to 28 in P2) was after about 20 seconds after the jam had started (happened immediately when selecting _UnStart_), the second (creating jams 30 to 45) was after a round of _UnStart Jam until I can't UnStart any more_, then entering TimeOut
![image](https://user-images.githubusercontent.com/28804763/164883586-3b0ded4a-631e-4741-8910-fe3b0aaad038.png)
I've attached the end-of-game JSON if it's any help
[STATS-0000-00-00_KingstonCityRollers_vs_KingstonCityRollers.json.txt](https://github.com/rollerderby/scoreboard/files/8546413/STATS-0000-00-00_KingstonCityRollers_vs_KingstonCityRollers.json.txt)

Bug 8a:
JSON output from IGRF page isn't populating with the date of the game in the file name
![image](https://user-images.githubusercontent.com/28804763/164884174-b7bb8cab-ff8c-4b0e-baea-c18216cb94cf.png)

Bug 8b & c:
  8b: IGRF _Update_ button doesn't appear to do anything?
  8c: XLSX statsbook fails to generate a file [ I had this issue in the Beta, too, but hadn't had time to push through something resembling a full game, so wasn't 100% if it was related to that ]
For both the above, see video below confirming that:
1. Game is marked as _Official Score_
2. There's a blank _Statsbook_ file in the base folder, and that the _Blank Statsbook File_ field is populated with the name of that file
3. When attempting to generate, getting **Failed - No File** as the browser download error [Google Chrome, 64-bit Version 100.0.4896.127 running on Windows 11 ; Also getting the same on MS Edge Versions 100.0.1185.44 and 100.0.1185.50]
https://user-images.githubusercontent.com/28804763/164884345-b6546b7b-242d-48a6-b76a-fbbb6decbf12.mp4

Bug 9:
In _Sheets_ the _Score + Lineups_ doesn't lineup side-by-side like in _Both Scores_ and _Both Lineups_. Looks like the attempt was there, but either the Score 'frame' is too low, or the Lineups 'frame' is too high
![image](https://user-images.githubusercontent.com/28804763/164885126-1cf3f78e-6bc1-4ebb-a856-e0a550618e97.png)

**Comment by frank-weinberg** [comment]
Thank you for collecting these and especially for the very helpful write up. I'll go through these over the weekend (and if need be the next week).

Just two questions:

- Was 7b also happening in conjunction with AutoStart Jam?
- For 8b/c is there any output from the backend? The symptoms look like statsbook generation is attempted but fails, which should result in output there. (I also noticed that you are clicking on "Download Statsbook" way too soon after clicking on "Update" - generating the statsbook takes 20-30s even on a fast computer. But presumably this was not your first try so if this was the only problem a file from a prior attempt should have been there.)

**Comment by bullseye555** [comment]
No worries @frank-weinberg (I'm an SQL Dev, and part of my job is to try and break other peoples code and tell them how I did it, and I get thoroughly annoyed when all I get from users is _hey, this doesn't work, fix it_ with no other context....)

- 7b - this only happened during AutoStart, I haven't tested with AutoStart off yet (I wanted to preserve the current environment incase you wanted to have a deeper look at it before it dropped off being a live game. I set the End of Game intermission to be something like 2.5billion hours... )
- 8b/c - Correct, it wasn't the first time I'd tried it; and no - there was no file generated, only the _failed_ attempted download that happens as soon as clicking on the _Download Statsbook_

**Comment by frank-weinberg** [comment]
No output on the backend at all? That's surprising.

And thinking about Bug 1 some more that also could have output on the backend. - From the screenshots it looks like you were using the teams from the import in Bug 2. (I'm thinking that maybe the import caused some inconsistency in team data that then in turn caused internal errors on start game & export when this team data was read. If so that should be reproducable with your attached data file.)

**Comment by bullseye555** [comment]
Ok - Didn't dig in the right folder tree (I was looking at my 4.1.4 folder instead of 5.0.0 for outputs!)
There's an XLSX file, but it's blank (eg, it's a standard statsbook file, renamed, but with no additional data from the game - I've attached for debugging purposes). For testing, I deleted and did a fresh _Download Statsbook_ but it didn't create this file - it was only when I attempted another _Update_ that this was created [filename not altered]
[~STATS-0000-00-00_KingstonCityRollers_vs_KingstonCityRollers.xlsx](https://github.com/rollerderby/scoreboard/files/8546649/STATS-0000-00-00_KingstonCityRollers_vs_KingstonCityRollers.xlsx)

Regarding Bug 1 & 2, I manually input the teams seen in the screenshots there. Bug 2 wasn't identified until after I'd run the mock game and was testing the Up/Download [I know there were issues with Download that have been resolved in an unreleased build, do I didn't lodge them again :D ]. I haven't tested Statsbook import as of yet, as I don't have one handy

**Comment by frank-weinberg** [comment]
So the blank "Flags" seen in the screenshot are from Bug 4a? (No number sounds like bench staff)

And regarding backend output: I was looking for log messages and/or backtraces in the Scoreboard.exe window.

**Comment by bullseye555** [comment]
Yeah, bugs 1, 3, and 4 are all interconnected and we're identified prior to starting the first jam

And the Scoreboard.exe window has nothing other that the start up data

**Comment by frank-weinberg** [comment]
@bullseye555 Got (most of) them (and a few more).

Only Bug 3 I didn't manage to reproduce fully. I did not find any way to get a "Not Skating" skater to show up in the button list. (I did get the ones with broken flags but that is just an expected side effect of Bugs 2 and 4.)

If you want to do further tests, I attached a zip that includes all fixes I have so far. Given that Bug 1 and the PT breakage reported by Kel on FB are somewhat severe, I'm currently aiming for a 5.0.1 release sometime next week. But if I know another larger test is imminent, I might push that back a bit.

[crg-scoreboard_v5.0.0-13-gc81daf42-20220424185747.zip](https://github.com/rollerderby/scoreboard/files/8550127/crg-scoreboard_v5.0.0-13-gc81daf42-20220424185747.zip)

Sidenote: Bugs 8 & 7 sent me down a rabbit hole of bugs and surprising behaviour that kept me busy for quite a few hours but at the bottom had a reward in the form of a massive speedup for statsbook exports (28s to 1-2s for your game on my computer).

**Comment by bullseye555** [comment]
Thanks Frank!

I was going to do some more testing this today anyway (as I hadn't done any
testing on the PLT/LT/PT screens)

I expected to have any further feedback for you in the next 12 hours or so
- I'll use the zip you provided to test with :)

I'll see if I can replicate Bug 3 again, with the exact steps for you

**Comment by bullseye555** [comment]
Ran this game 1st half with PT & 2x LTs, 2nd half with 2x PLT - no additional bugs found from those screens

> Bug 1

Confirmed resolved

> Bug 2

Confirmed resolved

> Bug 3

Unable to replicate, only Skaters showed (also confirmed numbers dropped off, and were added, when their flag was changed)

> Bug 4a & 4b

Both resolved

> Bug 5

Confirmed resolved

> Bug 6

Confirmed resolved

> Bug 7a

_AutoStart = Jam_ + _Unstart Jam_ still resulting in an endless loop of _UnStart Jam_ but lineup > 30sec, so Auto-Start Jam

> Bug 7b

Appears to be resolved - unable to get this to occur again

> Bug 8a

Confirmed resolved

> Bug 8b & 8c
> 8b

Clicking on _Update_ and leaving for ~2mins (without navigating away from page) doesn't change the _Last Updated_
![image](https://user-images.githubusercontent.com/28804763/165014671-77fa483f-db03-42cd-9d12-7ee165f50254.png)
![image](https://user-images.githubusercontent.com/28804763/165014881-32bdd47f-ebbf-43c7-987a-da4d461c2d0b.png)
> 8c

Still receiving the same error - _Failed - No File_ with no message in the scoreboard-Windows window
![image](https://user-images.githubusercontent.com/28804763/165014761-f3c96c88-4a6e-4568-8fdf-6a55f3c96b41.png)

Again, there's a _Statsbook_ file in **\html\game-data\xlsx** 
[~STATS-2022-04-25_KingstonCityRollers_vs_KingstonCityRollers.xlsx](https://github.com/rollerderby/scoreboard/files/8551357/STATS-2022-04-25_KingstonCityRollers_vs_KingstonCityRollers.xlsx)
, however this one has some data in it (and the SCORE sheets aren't calculating - looks like Excel is interpreting there being something in the _NI_ column, which it looks like there's a space ' '
![image](https://user-images.githubusercontent.com/28804763/165015400-5ee72093-cc48-423d-9d1d-615f89cc09b6.png)
![image](https://user-images.githubusercontent.com/28804763/165015450-680998d7-600d-4b16-89e2-43336d2f82dc.png)

If I **F2** into the cell (or otherwise click in, then out) it _then_ picks up that it's blank and the formula works
![image](https://user-images.githubusercontent.com/28804763/165015534-3dc278a0-7b1b-487e-b26e-73ad47a7d4ad.png)

> Bug 9

Confirmed resolved

**Comment by frank-weinberg** [comment]
Can you attach the json from the game that didn't export?

And 7a is weird - AutoStart didn't trigger again after Undo for me in the fixed version. But I'll have another look.

**Comment by bullseye555** [comment]
Attached

For clarification, it doesn't do the _UnStart_ a limited number of times like the 5.0.0 version, but in this version it just continuously loops through the _UnStart_/_AutoStart_

[STATS-2022-04-25_KingstonCityRollers_vs_KingstonCityRollers.json.txt](https://github.com/rollerderby/scoreboard/files/8552518/STATS-2022-04-25_KingstonCityRollers_vs_KingstonCityRollers.json.txt)

**Comment by bullseye555** [comment]
Sorry, to clarify on this point
> AutoStart didn't trigger again after Undo for me in the fixed version. But I'll have another look.

Explicit steps were:

1. AutoStart = Jam
![image](https://user-images.githubusercontent.com/28804763/165044064-c1d265a4-8060-4486-b536-bb523de9829b.png)
2. Lineup time hits 30 seconds and Jam auto-starts (at 32 seconds)
3. Un-Stop Jam
![image](https://user-images.githubusercontent.com/28804763/165044193-62e5a56d-5c32-4c38-89c9-ef11c14b14cd.png)
4. As _lineup >30_ and _AutoStart = Jam,_ automatically re-start the Jam

I just started a brand new instance to capture it (re-extracted the ZIP, started on a different port, no teams input, etc) and it worked as expected [UnStart reverted to Lineup and didn't re-start the Jam]. As I type, the _Lineup_ is up to 3:00 (+2:30 from original start) after I did the _UnStart_, so not sure why it did it in the previous instance

As I haven't been able to replicate it occurring again, happy to close it off

**Comment by frank-weinberg** [comment]
That last behavior is what is expected in this build. There is now a flag indicating that AutoStart has run which is only reset when StopJam/TO is triggered (either by button or AutoEnd).

I'll look into the export after work but given that the xlsx looks like it has all the data in it the problem may have been something windows specific on moving the temp file to the final file, which could be tough to debug as windows dumps the stack trace. (The formula thing also sounds like an Excel thing as LibreOffice (on Linux) does evaluate the formulas.)

**Comment by bullseye555** [comment]
Yeah, it's definitely an Excel thing. I'll go digging to see if I can find
and logs in the Event Viewer after dinner

**Comment by frank-weinberg** [comment]
I used my lunch break to generate a build where stack traces on statsbook export should go to the scoreboard-Windows window. Maybe you can run the export again with this build. (And if it works I'll convert other stack traces as well.)

[crg-scoreboard_v5.0.0-13-gc81daf42-dirty-20220425124804.zip](https://github.com/rollerderby/scoreboard/files/8553826/crg-scoreboard_v5.0.0-13-gc81daf42-dirty-20220425124804.zip)

**Comment by bullseye555** [comment]
Dropping screenshots in here.
I grabbed the JSON (Game-Data _and_ 0 Sec Ago), XLSX, blank statsbook, & Team images from the [crg-scoreboard_v5.0.0-13-gc81daf42-20220424185747.zip](https://github.com/rollerderby/scoreboard/files/8550127/crg-scoreboard_v5.0.0-13-gc81daf42-20220424185747.zip) build into this one (it threw a whole lot of missing file errors when I tried to load the JSONs without them!)

Initial load screen
![image](https://user-images.githubusercontent.com/28804763/165078266-5bcf77ae-a88f-4194-bb66-9a37fb03155c.png)
_Update_ attempted
![image](https://user-images.githubusercontent.com/28804763/165078376-b367c972-4790-415f-84b9-1f8242e03e4b.png)
(For reference, I don't have the file open anywhere)
When attempting _Download Statsbook_ it didn't add any additional info to the window

Deleted the XLSX file and re-attempted Update, only to get _File In Use_ again (Watching _File Explorer_, I can see the file gets created before the error gets thrown
![image](https://user-images.githubusercontent.com/28804763/165078700-3ece14f0-6909-4605-9586-11b4ea36d470.png)

See screen record below - in this instance, the XLSX file is still being flagged as _In Use_ by the windows file-system, even though I don't have it open, and it doesn't appear CRG does either. I'm heading to bed (wanted to try and get you this before I did), so can dig through Windows Events for you in the morning if you want me to see if I can see anything there
(Watching the vid in play-back, it looks like the _File In Use_ error shows up before the file is created)
https://user-images.githubusercontent.com/28804763/165079694-a35f5bf5-867d-4811-a2a4-5e0c32785308.mp4

**Comment by frank-weinberg** [comment]
I think I found it. I had forgotten that Java isn't RAII and thus you have to close files explicitly - only deleting the input/output stream isn't enough. That's fixed in the attached build.

I also found a function that marks an xlsx so that Excel will reevaluate all formulas on opening it and added that. So we should be down to zero known bugs. :)

[crg-scoreboard_v5.0.0-14-gff0e0b73-20220425171419.zip](https://github.com/rollerderby/scoreboard/files/8555840/crg-scoreboard_v5.0.0-14-gff0e0b73-20220425171419.zip)

**Comment by bullseye555** [comment]
Successful _Update_!
![image](https://user-images.githubusercontent.com/28804763/165193507-d37ad5b7-99b4-43c8-8b57-e9dc82db831f.png)
Successful _Download Statsbook_!
![image](https://user-images.githubusercontent.com/28804763/165193575-84b98405-3930-4a82-a651-b1d67306b606.png)

Sometimes I hate Excel... It's evaluating the **NI** cell as having a _zero-length string_ which is why the _ISBLANK_ is returning **FALSE**
![image](https://user-images.githubusercontent.com/28804763/165193865-ddbf474e-1b44-4f64-abc9-252ce5829c38.png)

[See this Microsoft Docs Article](https://docs.microsoft.com/en-us/office/troubleshoot/excel/isblank-function-return-false#:~:text=When%20you%20use%20the%20ISBLANK()%20function%20in%20a%20Microsoft,nothing%20is%20in%20the%20cell.)
In short, it's interpreting that there's _something_ in the cell, so ISBLANK(G4) will **always** return _FALSE_ until G4 has an edit attempted on the cell - when it then re-interrogates the contents of the cell, that's when it identifies it as empty and then ISBLANK(G4) will return _TRUE_ and the WFTDA's formula will calculate correctly

I don't suppose there's an **easy** way to update the _Jam Total_ formula to follow this logic when outputting the Statsbook? **=IF(OR(ISBLANK(A4),A4=""),"",IF(OR(ISBLANK(G4),G4=""),SUM(H4:P4),0))**
This will interpret ISBLANK **and** zero-length strings for the cell, and then the formula will calculate correctly
(Or a Bonus if you know someone at WFTDA that can change the base Statsbook formula 😄 )

**Comment by frank-weinberg** [comment]
> I don't suppose there's an easy way to update the Jam Total formula 

There is. But now that I knew to look for it I also found an easy way to mark a cell as blank. 

So I hope this file is as Excel likes it:

[STATS-2022-04-25_KingstonCityRollers_vs_KingstonCityRollers.xlsx](https://github.com/rollerderby/scoreboard/files/8588437/STATS-2022-04-25_KingstonCityRollers_vs_KingstonCityRollers.xlsx)

zip, if you want to do more testing:

[crg-scoreboard_v5.0.0-14-gff0e0b73-dirty-20220429075420.zip](https://github.com/rollerderby/scoreboard/files/8588443/crg-scoreboard_v5.0.0-14-gff0e0b73-dirty-20220429075420.zip)

**Comment by bullseye555** [comment]
Thanks Frank

Confirming Excel was happy with that - I also ran the output from the zip and all good

Thanks for clearing these up!

**Comment by frank-weinberg** [comment]
Yay. So 5.0.1 on 5/01.

**Comment by JeneralPain** [comment]
Unrelated, I just pulled the latest git clone dev and tried to ant build it out...

```
ubuntu@ip-172-26-11-146:~/scoreboard$ ant
Buildfile: /home/ubuntu/scoreboard/build.xml

-version:
    [mkdir] Created dir: /home/ubuntu/scoreboard/bin/com/carolinarollergirls/scoreboard/version
[propertyfile] Creating new property file: /home/ubuntu/scoreboard/bin/com/carolinarollergirls/scoreboard/version/release.properties

-manifest:

compile:
    [mkdir] Created dir: /home/ubuntu/scoreboard/html/images/teamlogo
    [mkdir] Created dir: /home/ubuntu/scoreboard/html/images/sponsor_banner
    [javac] Compiling 117 source files to /home/ubuntu/scoreboard/bin
     [copy] Copying 1 file to /home/ubuntu/scoreboard/bin
  [one-jar] Building jar: /home/ubuntu/scoreboard/lib/crg-scoreboard.jar

test-compile:
    [mkdir] Created dir: /home/ubuntu/scoreboard/test-bin
    [javac] Compiling 22 source files to /home/ubuntu/scoreboard/test-bin

test:
    [junit] Running com.carolinarollergirls.scoreboard.core.admin.ClientsImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.admin.ClientsImplTests
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.317 sec
    [junit] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.317 sec
    [junit] 
    [junit] ------------- Standard Error -----------------
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Unknown property Device(f7712fab-6b38-4438-9506-9d784e456574).NoSuchProperty()
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Could not get or create property Clients().Client(foo)
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] Unknown property Client(e501240b-8840-4ba5-9991-76ef519d2526).NoSuchProperty()
    [junit] ------------- ---------------- ---------------
    [junit] Running com.carolinarollergirls.scoreboard.core.admin.MediaImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.admin.MediaImplTests
    [junit] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.123 sec
    [junit] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.123 sec
    [junit] 
    [junit] Running com.carolinarollergirls.scoreboard.core.admin.SettingsImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.admin.SettingsImplTests
    [junit] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.05 sec
    [junit] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.05 sec
    [junit] 
    [junit] Running com.carolinarollergirls.scoreboard.core.game.BoxTripImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.game.BoxTripImplTests
    [junit] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.781 sec
    [junit] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.781 sec
    [junit] 
    [junit] Running com.carolinarollergirls.scoreboard.core.game.ClockImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.game.ClockImplTests
    [junit] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.825 sec
    [junit] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.825 sec
    [junit] 
    [junit] Running com.carolinarollergirls.scoreboard.core.game.FieldingImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.game.FieldingImplTests
    [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.52 sec
    [junit] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.52 sec
    [junit] 
    [junit] Running com.carolinarollergirls.scoreboard.core.game.GameImplTests
    [junit] Testsuite: com.carolinarollergirls.scoreboard.core.game.GameImplTests
    [junit] Tests run: 63, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 16.723 sec
    [junit] Tests run: 63, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 16.723 sec
    [junit] 
    [junit] Testcase: testJamStartUndoAfterPeriodEnd(com.carolinarollergirls.scoreboard.core.game.GameImplTests):       FAILED
    [junit] expected:<0> but was:<1>
    [junit] junit.framework.AssertionFailedError: expected:<0> but was:<1>
    [junit]     at com.carolinarollergirls.scoreboard.core.game.GameImplTests.tearDown(GameImplTests.java:106)
    [junit]     at jdk.internal.reflect.GeneratedMethodAccessor5.invoke(Unknown Source)
    [junit]     at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 
    [junit] 

BUILD FAILED
/home/ubuntu/scoreboard/build.xml:147: Test com.carolinarollergirls.scoreboard.core.game.GameImplTests failed

Total time: 35 seconds
```

**Comment by frank-weinberg** [comment]
@JeneralPain  I can't reproduce that with either dev or the branch that has the fixes from this ticket.
Does this happen consistently (at the same test) for you? What JDK? What version of junit?
